framework/src/modules/chain/bft/utils.js

```diff
@@ -26,6 +35,38 @@ const { validate } = require('../../../../src/controller/validator');
 const validateBlockHeader = blockHeader =>
 	validate(blockHeaderSchema, blockHeader);
 
+/**
+ * Perform all checks outlined in Step 1 of the section "Processing Blocks" except for checking height, parentBlockID and delegate slot (block B may be in the future and assume different delegates that are not active in the round of block A). If any check fails, the peer that sent block B is banned and the node aborts the process of moving to a different chain.
+ *
+ * https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#moving-to-a-different-chain
+ *
+ * @param lastBlock
+ * @param block
+ */
+const verifyBlockForChainSwitching = (lastBlock, block) => {
```

Comment:
probably this is not responsibility of bft

Reply:
Its BFT to decide for chain switching, so bft have its own validation to do it.